codegen/scala-gen/src/test/scala/com/akkaserverless/codegen/scalasdk/EventSourcedEntitySourceGeneratorSuite.scala

```diff
@@ -0,0 +1,179 @@
+         |    effects.error("The command handler for `Get` is not implemented, yet")
+         |
+         |  override def setEvent(currentState: MyState, setEvent: SetEvent): MyState =
+         |    throw new RuntimeException("The event handler for `SetEvent` is not implemented, yet")
+         |}
+         |""".stripMargin)
+  }
+  test("it can generate an abstract event sourced entity implementation") {
+    val str = abstractEntity(testData.eventSourcedEntity(), testData.simpleEntityService()).content
+    assertNoDiff(
+      str,
+      s"""|package com.example.service.domain
+          |
+          |import com.akkaserverless.scalasdk.eventsourcedentity.EventSourcedEntity
+          |import com.example.service
+          |import com.external.Empty
+          |
+          |// This code is managed by Akka Serverless tooling.
+          |// It will be re-generated to reflect any changes to your protobuf definitions.
+          |// DO NOT EDIT
+          |
+          |/** An event sourced entity. */
+          |abstract class AbstractMyEntity extends EventSourcedEntity[MyState] {
+          |
+          |  def set(currentState: MyState, setValue: service.SetValue): EventSourcedEntity.Effect[Empty]
+          |
+          |  def get(currentState: MyState, getValue: service.GetValue): EventSourcedEntity.Effect[service.MyState]
+          |
+          |  def setEvent(currentState: MyState, setEvent: SetEvent): MyState
+          |}
+          |
+      """.stripMargin)
+  }
+
+  test("it can generate an event sourced entity handler implementation") {
+    val str = handler(testData.eventSourcedEntity(), testData.simpleEntityService()).content
+    assertNoDiff(
+      str,
+      s"""|package com.example.service.domain
+          |
+          |import com.akkaserverless.javasdk.impl.eventsourcedentity.EventSourcedEntityHandler.CommandHandlerNotFound
+          |import com.akkaserverless.javasdk.impl.eventsourcedentity.EventSourcedEntityHandler.EventHandlerNotFound
+          |import com.akkaserverless.scalasdk.eventsourcedentity.CommandContext
+          |import com.akkaserverless.scalasdk.eventsourcedentity.EventSourcedEntity
+          |import com.akkaserverless.scalasdk.impl.eventsourcedentity.EventSourcedEntityHandler
+          |import com.example.service
+          |
+          |// This code is managed by Akka Serverless tooling.
+          |// It will be re-generated to reflect any changes to your protobuf definitions.
+          |// DO NOT EDIT
+          |
+          |/**
+          | * An event sourced entity handler that is the glue between the Protobuf service <code>CounterService</code>
+          | * and the command handler methods in the <code>Counter</code> class.
+          | */
+          |class MyEntityHandler(entity: MyEntity) extends EventSourcedEntityHandler[MyState, MyEntity](entity) {
+          |  def handleEvent(state: MyState, event: Any): MyState = {
+          |    event match {
+          |      case SetEvent =>
```

Comment:
that looks strange, matching on the companion object?

shouldn't it be `case setEvent: SetEvent`, and then the asInstanceOf isn't needed?

Reply:
yes fixed (I merged the fix in now)